static/extensions/MubiLop/penguinhook.js

```diff
@@ -0,0 +1,258 @@
+            const message = Scratch.Cast.toString(MESSAGE);
+            const webhookUrl = Scratch.Cast.toString(WEBHOOK);
+            let nameReq;
+
+            if (name) {
+                nameReq = name;
+            }
+
+            const payload = {
+                username: nameReq,
+                avatar_url: imageUrl,
+                content: message
+            };
+
+            console.log(JSON.stringify(payload))
+
+            return fetch(webhookUrl, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(payload)
+            })
+                .then(response => {
+                    if (!response.ok) {
+                        console.error('Webhook request failed:', response.statusText);
+                        return 'Webhook request failed';
+                    }
+                    return 'Webhook sent successfully';
+                })
+                .catch(error => {
+                    console.error('Error sending webhook:', error);
+                    return 'Error sending webhook';
+                });
+        }
+
+        sendWebhookWithEmbed(args) {
+            const embed = {};
+            const name = {};
+
+            if (this.embedProperties.title) {
+                embed.title = Scratch.Cast.toString(this.embedProperties.title);
+            }
+
+            if (this.embedProperties.description) {
+                embed.description = Scratch.Cast.toString(this.embedProperties.description);
+            }
+
+            if (this.embedProperties.footer) {
+                embed.footer = { text: Scratch.Cast.toString(this.embedProperties.footer) };
+            }
+
+            if (this.embedProperties.name) {
+                name.name = Scratch.Cast.toString(this.embedProperties.name);
+            }
+
+            const webhookUrl = Scratch.Cast.toString(args.WEBHOOK);
+            const payload = {
+                embeds: [embed],
+                username: name.name,
+            };
+
+            return fetch(webhookUrl, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(payload)
+            })
+                .then(response => {
+                    if (!response.ok) {
+                        console.error('Webhook request with embeds failed:', response.statusText);
+                        return 'Webhook request with embeds failed';
+                    }
+                    return 'Webhook with embeds sent successfully';
+                })
+                .catch(error => {
+                    console.error('Error sending webhook with embeds:', error);
+                    return 'Error sending webhook with embeds';
+                });
+        }
+
+        sendjson(args) {
+            try {
+                return fetch(args.WEBHOOK, {
```

Comment:
Scratch.canFetch() is not checked before this, meaning this will just send data to a random url without the user knowing at all